var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -17,9 +17,12 @@ class Esmf(MakefilePackage):
 
     homepage = "https://www.earthsystemcog.org/projects/esmf/"
     url = "https://github.com/esmf-org/esmf/archive/ESMF_8_0_1.tar.gz"
+    git = "https://github.com/esmf-org/esmf/"
 
     maintainers = ["climbfuji"]
 
+    version("develop", branch="develop")
```

Comment:
I think this will not work since the patches are version dependent and at least the gcc patch is failing in my case because it is not needed for ESMF 8.4.0 versions and snapshots. So, we need to find a way to cover those cases.

Reply:
Are you saying that the gcc patch from this line:
      patch("gcc.patch", when="@:7.0 %gcc@6:")   
is being applied in develop?   That should not happen.

Reply:
Yes. It will apply to develop and it will fail. At least it was working like that in my case. It was failing because that part of line is already fixed in ESMF. But I think issue is more general in here, it seems that `when` argument in the patch definition is not working correctly with the branches (and maybe with tags which are automatically added - I have away to do it) due to the missing version information.

Reply:
As I understand it "develop" is considered > any named version.   But I will test with gcc and try to resolve.

Reply:
Maybe this is not a fix for package.py and needs to be fixed in the spack in more general way. @alalazo Do you have suggestion or comments about it?